tools/aiecc/aiecc.cpp

```diff
@@ -0,0 +1,862 @@
+// NOTE: requires chess
+//cl::opt<bool> AIESim("aiesim", cl::desc("Generate aiesim Work folder"));
+cl::opt<std::string> Peano("peano", cl::desc("Root directory where peano compiler is installed"));
+cl::opt<bool> Compile("compile", cl::desc("Enable compiling of AIE code"), cl::init(true));
+cl::opt<bool> Link("link", cl::desc("Enable compiling of AIE code"), cl::init(true));
+cl::opt<std::string> HostArch("host-target", cl::desc("Target architecture of the host program"), cl::init(HOST_ARCHITECTURE));
+cl::opt<bool> HostCompile("compile-host", cl::desc("Enable compiling of the host program"), cl::init(true));
+cl::list<std::string> HostArgs(cl::Positional, cl::desc("[<host compiler arg>]..."));
+cl::opt<unsigned> Threads("j", cl::desc("Compile with max n-threads in the machine (default is 4).  An argument of zero corresponds to the maximum number of threads on the machine."), cl::init(4));
+cl::opt<bool> Profiling("profile", cl::desc("Profile commands to find the most expensive executions."));
+cl::opt<bool> Unified("unified", cl::desc("Compile all cores together in a single process"));
+cl::opt<bool> DryRun("n", cl::desc("Disable actually executing any commands."));
+cl::opt<bool> Progress("progress", cl::desc("Show progress visualization."));
+cl::opt<bool> GenerateIPU("aie-generate-ipu", cl::desc("Generate ipu instruction stream"));
+cl::opt<bool> OnlyGenerateIPU("aie-only-generate-ipu", cl::desc("Generate ipu instruction stream only"));
+cl::opt<std::string> IPUInstsName("ipu-insts-name", cl::desc("Output instructions filename for IPU target"), cl::init("ipu_insts.txt"));
+cl::opt<std::string> XCLBinName("xclbin-name", cl::desc("Output xclbin filename for CDO/XCLBIN target"), cl::init("final.xclbin"));
+cl::opt<std::string> XCLBinKernelName("xclbin-kernel-name", cl::desc("Kernel name in xclbin file"), cl::init("MLIR_AIE"));
+cl::opt<std::string> XCLBinInstanceName("xclbin-instance-name", cl::desc("Instance name in xclbin metadata"), cl::init("MLIRAIE"));
+cl::opt<std::string> XCLBinKernelID("xclbin-kernel-id", cl::desc("Kernel id in xclbin file"), cl::init("0x901"));
+
+void addLowerToLLVMPasses(OpPassManager &pm) {
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
+  pm.addPass(createConvertVectorToLLVMPass());
+  pm.addPass(memref::createExpandStridedMetadataPass());
+  pm.addPass(createLowerAffinePass());
+  pm.addPass(createConvertMathToLLVMPass());
+  pm.addPass(createArithToLLVMConversionPass());
+  pm.addPass(createFinalizeMemRefToLLVMConversionPass());
+  pm.addPass(createConvertFuncToLLVMPass({.useBarePtrCallConv=true}));
+  pm.addPass(createConvertControlFlowToLLVMPass());
+  pm.addPass(createCanonicalizerPass());
+  pm.addPass(createCSEPass());
+}
+
+struct Toolkit {
+  std::string TargetArch;
+  sfs::path PeanoDir;
+  sfs::path InstallDir;
+  sfs::path AIEToolsDir;
+  sfs::path TempDir;
+};
+
+int runTool(StringRef Program, ArrayRef<StringRef> Args,
```

Comment:
It seems like the rest of the code is significantly complicated by having this take StringRefs...

Reply:
That's what is accepted by the LLVM helper function I'm using.

Reply:
I changed it to std::strings.